### PR TITLE
For #22832 - Added parameters to base Text composables

### LINKED ISSUE #22832: Add fontFamily and letterSpacing parameters to root Text Composables

### PR DESCRIPTION
**Pull request by MozillaNoah**
Fixes #22832 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/compose/Text.kt [hunk 1]
```diff
@@ -16,21 +18,28 @@ import org.mozilla.fenix.theme.FirefoxTheme
  * @param text The text to be displayed.
  * @param modifier [Modifier] to apply to this layout node.
  * @param fontSize The size of glyphs to use when painting the text.
+ * @param fontFamily The font family to be used when rendering the text. See [TextStyle.fontFamily].
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/Text.kt:21]:
```suggestion
 * @param fontFamily The font family to be used when rendering the text.
```
I think we dropped this the last time we chatted adding these initial composable functions.

File: app/src/main/java/org/mozilla/fenix/compose/Text.kt [hunk 2]
```diff
@@ -40,20 +49,27 @@ fun PrimaryText(
...
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/Text.kt:52]: Same as above.

File: app/src/main/java/org/mozilla/fenix/compose/Text.kt [hunk 3]
```diff
@@ -40,20 +49,27 @@ fun PrimaryText(
...
  * @param maxLines An optional maximum number of lines for the text to span.
  */
+@Suppress("LongParameterList")
 @Composable
 fun SecondaryText(
     text: String,
     modifier: Modifier = Modifier,
     fontSize: TextUnit,
+    fontFamily: FontFamily? = null,
+    letterSpacing: TextUnit = TextUnit.Unspecified,
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/Text.kt:63]: Let's also add `overflow` as well.

**Comment by mcarare** [comment]
Just a thought here: Could we further formalize these using styles as we did in [Focus](https://github.com/mozilla-mobile/focus-android/blob/2d8565882404bfc6f38543149d9457a03b7ee6fb/app/src/main/java/org/mozilla/focus/ui/theme/FocusTypography.kt)?

Instead of passing font size, font family, letter-spacing, etc,  just passing a predefined style for each component?

**Comment by gabrielluong** [comment]
> Just a thought here: Could we further formalize these using styles as we did in [Focus](https://github.com/mozilla-mobile/focus-android/blob/2d8565882404bfc6f38543149d9457a03b7ee6fb/app/src/main/java/org/mozilla/focus/ui/theme/FocusTypography.kt)?
> 
> Instead of passing font size, font family, letter-spacing, etc, just passing a predefined style for each component?

This does look very neat. I agree with further formalization using text styles to reflect the component. I think it will require some effort and investigation to the current text components we have in Compose and refactoring those as well. I would suggest maybe as a follow up to not block the tabs tray compose work.

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**